File: goapi/internal/domain/ingredient_nutrient.go
package domain

import "time"

type IngredientNutrient struct {
    ID                  uint      `json:"id" gorm:"primaryKey"`
    IngredientID        uint      `json:"ingredient_id"`
    NutrientID          uint      `json:"nutrient_id"`
    ContentQuantity     int       `json:"content_quantity"`
    ContentUnit         string    `json:"content_unit"`
    ContentUnitPer      int       `json:"content_unit_per"`
    ContentUnitPerUnit  string    `json:"content_unit_per_unit"`
    CreatedAt           time.Time `json:"created_at"`
    UpdatedAt           time.Time `json:"updated_at"`
}

File: graphqlapi/src/graphql/types/prisma/IngredientNutrient.ts
import { ObjectType, Field, ID } from 'type-graphql';

@ObjectType()
export class IngredientNutrient {
  @Field(() => ID)
  id!: string;

  @Field(() => ID, { nullable: true })
  ingredientId?: string;

  @Field(() => ID, { nullable: true })
  nutrientId?: string;

  @Field({ nullable: true })
  contentQuantity?: number;

  @Field({ nullable: true })
  contentUnit?: string;

  @Field({ nullable: true })
  contentUnitPer?: number;

  @Field({ nullable: true })
  contentUnitPerUnit?: string;

  @Field()
  createdAt!: Date;

  @Field()
  updatedAt!: Date;
}
